api/app/graphql/mutations/update_task.rb

```diff
@@ -5,9 +5,15 @@ class UpdateTask < BaseMutation
     argument :task_id, ID, required: true, loads: Types::TaskType
     argument :name, String, required: true
     argument :complete, Boolean, required: true
+    argument :user_id, ID, required: true
+    argument :duedate, GraphQL::Types::ISO8601Date, required: false
 
-    def resolve(task:, name:, complete:)
-      task.update!(name: name, complete: complete)
+    def resolve(task:, name:, complete:, user_id:, duedate:)
+      if task.complete != complete && complete == true
+        NotifyTeamLeaderMailer.notify_team_leader(task.id, user_id).deliver
+        task.update!(name: name, complete: complete, user_id: user_id, duedate: duedate)
```

Comment:
I'm not sure you need to `update!` the task inside the if statement. Its being done just after no matter what.

Reply:
Yea 100% agree with you! I actually didn't have this line here originally, and only had the `update!` on line 16, but then when I tested manually it it didn't seem to update even though the email was sent, so I added the `update!` to both places. If possible I'd love to debug together this so I can know what is happening.